### PR TITLE
Add additional error handling to unary RPCs

### PR DESCRIPTION
**Pull request by theacodes**

### CODE REVIEW

**Comment by tseaver** [comment]
Maybe the commit message should also note the switch away from the root logger?

File: pubsub/google/cloud/pubsub_v1/subscriber/_protocol/bidi.py [hunk 1]
```diff
@@ -481,6 +481,11 @@ def _thread_main(self):
                 '%s caught unexpected exception %s and will exit.',
                 _BIDIRECTIONAL_CONSUMER_NAME, exc)
 
+        else:
```
Comment (crwilcox) [inline pubsub/google/cloud/pubsub_v1/subscriber/_protocol/bidi.py]: Is self._bidi_rpc.is_active being false really unexpected?

Comment (theacodes) [reply pubsub/google/cloud/pubsub_v1/subscriber/_protocol/bidi.py]: Yep, it really shouldn't happen. This is to help produce more logs to debug.

Comment (crwilcox) [reply pubsub/google/cloud/pubsub_v1/subscriber/_protocol/bidi.py]: SGTM. I just didn't know if 'unexpectedly' was a bit strong or not.

**Comment by theacodes** [comment]
> Maybe the commit message should also note the switch away from the root logger?

Eh, too minor to call out.

**Pull request merged**